plugins/Tools.cc

```diff
@@ -339,6 +339,76 @@ void HHAnalyzer::matchOfflineLepton(const HLTProducer& hlt, HH::Dilepton& dilept
     }
 }
 
+float HHAnalyzer::getL1TPhi(int charge, const LorentzVector& p) {
+    float pt = p.Pt();
+    float phi = p.Phi();
+    float theta = p.Theta();
+    return phi + charge * (1. / pt) * (10.48 - 5.1412 * theta + 0.02308 * theta * theta);
+}
+
+bool HHAnalyzer::sameEndCap(const LorentzVector& p1, const LorentzVector& p2) {
+    return p1.Eta() * p2.Eta() > 0 && std::abs(p1.Eta()) > 1.24 && std::abs(p2.Eta()) > 1.24;
+}
+
+float HHAnalyzer::translatePhi(float phi, float translation/*=0*/) {
+    phi += translation; // translate
+    phi = std::fmod(phi, 2 * M_PI); // put between -2pi, 2pi
+    phi = (phi > 0) ? phi : (2 * M_PI + phi); // put between 0, 2pi
+    return phi;
+}
+
+int HHAnalyzer::getPhiSector(float phi, float start, float end) {
+    for (int i = 0; i < 6; i++) {
+        if (start + i * M_PI / 3 <= phi && phi < end + i * M_PI / 3)
+            return i;
+    }
+    return -1;
+}
+
+bool HHAnalyzer::isCSCSameSector(const Lepton& lep1, const Lepton& lep2) {
+    if (!sameEndCap(lep1.p4, lep2.p4))
+        return false;
+
+    float phi1 = translatePhi(getL1TPhi(lep1.charge, lep1.p4));
+    float phi2 = translatePhi(getL1TPhi(lep2.charge, lep2.p4));
+
+    int sector1 = getPhiSector(phi1, 15*M_PI/180, 65*M_PI/180);
+    int sector2 = getPhiSector(phi2, 15*M_PI/180, 65*M_PI/180);
+
+    int overlap1 = getPhiSector(phi1, 5*M_PI/180, 15*M_PI/180);
+    int overlap2 = getPhiSector(phi2, 5*M_PI/180, 15*M_PI/180);
+
+    if (sector1 >= 0 && sector2 >= 0 && sector1 == sector2)
+        return true;
+
+    if (overlap1 >= 0 && overlap2 >= 0 && overlap1 == overlap2)
+        return true;
```

Comment:
And actually reading the twiki again, it's not even clear what to do in case 2) for this type of situation. But anyway as I said these will be thrown away by the other function...